path_login.go

```diff
@@ -350,8 +350,12 @@ func (b *kubeAuthBackend) parseAndValidateJWT(ctx context.Context, client *http.
 		}
 	}
 
-	// validate the audience if the role expects it
-	if role.Audience != "" {
+	// Roles will need to specify an audience in Vault v1.21+.
+	// Log a warning if the role does not specify one.
+	if strings.TrimSpace(role.Audience) == "" {
+		b.Logger().Warn("Role %s was used to authenticate into Vault. It does not have an audience specified. "+
+			"Vault v1.21+ will require an audience to be specified in the role to authenticate successfully.", roleName)
```

Comment:
This is incorrect usage of the Warn method. Everything after the first argument must be key/value pairs.